classes/core/PKPEventServiceProvider.php

```diff
@@ -39,22 +39,28 @@ class PKPEventServiceProvider extends EventServiceProvider
      */
     protected $subscribe = [];
 
+    private static function getCacheKey(): string
+    {
+        return __METHOD__ . static::MAX_CACHE_LIFETIME;
+    }
+
+    public static function clearCache(): void
+    {
+        Cache::forget(static::getCacheKey());
+    }
+
     /**
      * Get the discovered events and listeners for the application
      *
      * @return array
      */
     public function getEvents()
     {
-        $cacheManager = CacheManager::getManager();
-        /** @var FileCache */
-        $cache = $cacheManager->getCache('event', PKPApplication::CONTEXT_SITE, function (FileCache $cache) {
-            $cache->setEntireCache($this->discoveredEvents());
-        });
-
+        $expiration = DateInterval::createFromDateString(static::MAX_CACHE_LIFETIME);
+        $events = Cache::remember(static::getCacheKey(), $expiration, fn () => $this->discoveredEvents());
```

Comment:
Do we still need this workaround after adopting Laravel cache or can we delegate it back to the Container?

Reply:
Here I wanted to inherit this class from the `Illuminate\Foundation\Support\Providers` instead of `Illuminate\Events\EventServiceProvider` and just override with customizations.

But I think it's worth to get this fix into 3.4.0, so I played safe and it was just a quick fix for a problem that I found while working in another task.

Reply:
FYI I've just checked the implementation, and it's a kind of "hard" cache, which is created through an "artisan" command, so I'll keep the customization.